pyirf/interpolation/component_estimators.py

```diff
@@ -540,8 +563,101 @@ def __call__(self, target_point):
             effective areas. For RAD_MAX_2D of shape (n_energy_bins, n_fov_offset_bins)
 
         """
+        # First, construct estimation without handling fill-values
+        full_estimation = super().__call__(target_point)
+        # Safeguard against extreme extrapolation cases
+        full_estimation[full_estimation < 0] = 0
+
+        # Early exit if fill_values should not be handled
+        if not self.fill_val:
+            return full_estimation
+
+        # Raise error if fill-values should be handled in >=3 dims
+        if self.grid_dim >= 3:
+            raise ValueError(
+                "Fill-value handling only supported in up to two grid dimensions."
+            )
+
+        # Early exit if a nearest neighbor estimation would be overwritten
+        if self.grid_dim == 1:
+            if (
+                (target_point < self.grid_points.min())
+                or (target_point > self.grid_points.max())
+            ) and issubclass(self.extrapolator.__class__, BaseNearestNeighborSearcher):
+                return full_estimation
+            elif issubclass(self.interpolator.__class__, BaseNearestNeighborSearcher):
+                return full_estimation
+        elif self.grid_dim == 2:
+            target_simplex = self.triangulation.find_simplex(target_point)
+
+            if (target_simplex == -1) and issubclass(
```

Comment:
You are checking for `BaseNearestNeighborSearcher` here twice, only the check on `target_simplex` is added, but then you do the same in both cases.

So just checking for `issubclass(BaseNearestNeighborSearcher` should be enough?

Reply:
`target_simplex == -1` means `target_point` outside grid, thus this checks for cases, where extrapolation will actually be applied and only in these cases the `issubclass` check on `self.extrapolator.__class__` yields the needed information. In the second case, thus `target_simplex != -1`, this needs to check `self.interpolator.__class__`.

Basically, this accounts for the possibility that the user mixes and uses NearestNeighbor for extrapolation and actual interpolation otherwise (and vice-versa) at the same time.